### PR TITLE
Fix core and test warning for MSVC

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: sdk/core/core/test/main.c [hunk 1]
```diff
@@ -660,7 +660,7 @@ int main() {
     TEST_ASSERT(az_span_is_equal(az_span_builder_result(&builder), uri_decoded));
   }
   {
-    int16_t const url_max = 100;
+    // int16_t const url_max = 100;
```
Comment (gilbertw) [inline sdk/core/core/test/main.c]: remove rather than comment

File: sdk/core/core/CMakeLists.txt [hunk 2]
```diff
@@ -73,7 +73,10 @@ add_executable (
     test/test_json_builder.c
     test/test_json_get_by_pointer.c
     test/test_json_pointer.c
-    test/test_json_string.c  
+    test/test_json_string.c
+    test/test_pair_span.c
```
Comment (barcharcraz) [inline sdk/core/core/CMakeLists.txt]: this seems like an unrelated change.
